### ISSUE TITLE
Rule:  string type given to static assert first parameter

### ISSUE THREAD

**Issue by ljmf00-wekaio** [ljmf00/ldclint#64]
Code smell:
```
static assert("unexpected object type");
```
Expected:
```
static assert(false, "unexpected object type");
```

Suggested by @EyalIO

**Comment by EyalIO** [comment]
Thanks :)

**Comment by JohanEngelen** [comment]
Note: from LDC 1.37 onwards, this triggers a deprecation, so no need for this in the linter after we upgrade to 1.37
https://d.godbolt.org/z/hbTs8WeEq

**Comment by ljmf00-wekaio** [comment]
I'll keep it here, if the plugin supports older versions, but thanks for getting to know that!
